### ISSUE TITLE
Incorrect behavior of Call mediator in blocking mode

### ISSUE THREAD

**Issue by KasunAratthanage** [wso2/product-ei#3015]
**Description:**
When a SOAP message receives over a proxy service and deliver over HTTP endpoint via call mediator in blocking mode, the given request content message return as the response message.

**Suggested Labels:**
Bug[Product-excellence]

**Suggested Assignees:**

**Affected Product Version:**
wso2ei-6.4.0

**OS, DB, other environment details and versions:**    
OS - Ubuntu

**Steps to reproduce:**
1. Create a HTTP endpoint.
2. Add proxy service.
3. Add call mediator with blocking mode with respond mediator.
4. Send a SOAP message request.

Example proxy service 

```xml
<?xml version="1.0" encoding="UTF-8"?>
<proxy xmlns="http://ws.apache.org/ns/synapse"
       name="2_1_1_2_HttpMessageWithoutAlteringDownstreamTestProxy"
       startOnLoad="true"
       statistics="disable"
       trace="disable"
       transports="http,https">
   <target>
      <inSequence>
         <call blocking="true">
            <endpoint key="HttpMessageWithoutAlteringDownstreamTestHttpEP"/>
         </call>
         <respond/>
      </inSequence>
      <outSequence/>
      <faultSequence/>
   </target>
   <description/>
</proxy>
 ```
**Related Issues:**

**Comment by Vathsan** [comment]
Fixed in [#1301](https://github.com/wso2/wso2-synapse/pull/1301)

**Comment by sdkottegoda** [comment]
Resolving since https://github.com/wso2/wso2-synapse/pull/1301 is merged
